Entities/Activity.cpp

```diff
@@ -0,0 +1,1136 @@
+		if (m_ControlledActor[player]) {
+			m_ControlledActor[player]->SetControllerMode(Controller::CIM_PLAYER, player);
+
+			// Play actor switching sound effects
+			(m_ControlledActor[player] == m_Brain[player]) ? g_GUISound.BrainSwitchSound()->Play() : g_GUISound.ActorSwitchSound()->Play();
+			// Only play air swoosh if actors are out of sight of each other
+			if (prevActor && Vector(prevActor->GetPos() - m_ControlledActor[player]->GetPos()).GetMagnitude() > g_FrameMan.GetResX() / 2) { g_GUISound.CameraTravelSound()->Play(); }			
+
+			ReassignSquadLeader(player, team);
+		}
+
+		// Follow the new guy normally
+		m_ViewState[player] = NORMAL;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void Activity::EnteredOrbit(Actor *orbitedCraft) {
+		if (!orbitedCraft) {
+			return;
+		}
+		int team = orbitedCraft->GetTeam();
+
+		// Find out cost multipliers for metagame, or we end up AI's earning money when their craft return if their nativeCostMultiplier is too low
+		float foreignCostMult = 1.0F;
+		float nativeCostMult = 1.0F;
+		if (g_MetaMan.GameInProgress()) {
+			for (int player = 0; player < MAXPLAYERCOUNT; player++) {
+				if (GetTeamOfPlayer(player) == orbitedCraft->GetTeam()) {
+					const MetaPlayer *metaPlayer = g_MetaMan.GetMetaPlayerOfInGamePlayer(player);
+					if (metaPlayer) {
+						foreignCostMult = metaPlayer->GetForeignCostMultiplier();
+						nativeCostMult = metaPlayer->GetNativeCostMultiplier();
+					}
+				}
+			}
+		}
+
+		// Did a brain just evacuate the Scene??
+		bool brainOnBoard = orbitedCraft->HasObjectInGroup("Brains");
+		// Total value of ship and cargo and crew
+		float totalValue = orbitedCraft->GetTotalValue(0, foreignCostMult, nativeCostMult);
+		// TODO ARGH WHAT IF TWO PLAYERS ON SAME TEAM ARE OF DIFFERENT TECHSS??!?!?!?$?!?!$?!$?
+		// A: Just let the base cost be the liquidation value.. they could cheat otherwise, one buying, one selling
+		char str[64];
+		sprintf_s(str, sizeof(str), "Returned Craft + Cargo added %.0f oz to Funds!", totalValue);
+		for (int player = 0; player < MAXPLAYERCOUNT; ++player) {
+			if (m_IsActive[player]) {
+				// Figure out whose brain just left the building
+				if (brainOnBoard && orbitedCraft == GetPlayerBrain(player)) {
+					m_BrainEvacuated[player] = true;
+					sprintf_s(str, sizeof(str), "YOUR BRAIN HAS BEEN EVACUATED BACK INTO ORBIT!");
+				}
+
+				if (m_Team[player] == team) {
+					g_FrameMan.ClearScreenText(ScreenOfPlayer(ScreenOfPlayer(player)));
+					g_FrameMan.SetScreenText(str, ScreenOfPlayer(player), 0, 3500);
+					m_MsgTimer[player].Reset();
+				}
+			}
+		}
+
+		m_TeamFunds[team] += totalValue;
+		orbitedCraft->SetGoldCarried(0);
+		orbitedCraft->SetHealth(orbitedCraft->GetMaxHealth());
+
+		// Counter-adjust the death toll because the craft leaving (being deleted) will increment
```

Comment:
I'd rewrite this a little so it's more worded around explaining why we're doing this, i.e.
`// The craft entering orbit will count as a death for the team because it's being deleted, so we need to decrement the team's death count to keep it correct.`